app/src/main/scala/Query.scala

```diff
@@ -0,0 +1,46 @@
+package roach
+
+import scala.scalanative.unsafe.Zone
+import scala.util.NotGiven
+
+opaque type Query[T] = T => (Database, Zone) ?=> roach.Result
```

Comment:
If it works out - move it to roach

Using this instead of the prepared statements, that require a state to be held in the connection, and that state being rehydrated when Pool restarts the connection.

Not worth the effort really, until Roach supports pool-level prepared statements.